### PR TITLE
Openssl in coreutils makes *sum programs much faster (RFC)

### PR DESCRIPTION
**Pull request by viric**
sha256sum, md5sum,. run much faster. Factor 2x to 4x.

You can evaluate that by comparing the cpu time of your 'time sha256sum bigfile' and 'time openssl sha256 bigfile'.

What do you think? I use these a lot, and I guess that git-annex users too.

### CODE REVIEW

**Comment by FRidh** [comment]
@GrahamcOfBorg eval

File: pkgs/tools/misc/coreutils/default.nix [hunk 1]
```diff
@@ -1,5 +1,6 @@
 { stdenv, lib, buildPackages
 , autoreconfHook, texinfo, fetchurl, perl, xz, libiconv, gmp ? null
+, openssl ? null
```
Comment (matthewbauer) [inline pkgs/tools/misc/coreutils/default.nix]: You shouldn't need the default. openssl will always be defined in a callPackage.

**Comment by viric** [comment]
Ok. I'll merge it when these automatic tests finish.

**Comment by Mic92** [comment]
Why was this merged to master instead of staging?

**Comment by viric** [comment]
@dtzWill @mic92 my fault - I didn't think about staging and noone complained on the ->master PR so then I went on.

Why did you say that "openssl is not allowed in stdenv"?

**Comment by Mic92** [comment]
coreutils is part of stdenv. To build stdenv there is a pre-build bootstrap tarball. To use openssl in stdenv it also needs to be part of the boostraping, I think.

**Comment by LnL7** [comment]
https://hydra.nixos.org/build/78481294/nixlog/1/tail

**Comment by viric** [comment]
would a stdenv.isLinux be enough? I don't know what's that Darwin condition.

**Pull request merged**